src/common/hooks/analytics/use-analytics.ts

```diff
@@ -0,0 +1,51 @@
+import { useCurrentNetworkState } from '@store/network/networks.hooks';
+import { useLocation } from 'react-router-dom';
+import { IS_TEST_ENV } from '@common/constants';
+import { EventParams, PageParams } from '@segment/analytics-next/dist/pkg/core/arguments-resolver';
+import { analytics } from '@common/segment-init';
+import { logger } from '@common/logger';
+
+const IGNORED_PATH_REGEXPS = [/^\/$/];
+
+function isIgnoredPath(path: string) {
+  return IGNORED_PATH_REGEXPS.find(regexp => regexp.test(path));
+}
+
+export function useAnalytics() {
+  const currentNetwork = useCurrentNetworkState();
+  const using_default_hiro_api = /^https:\/\/.*\.stacks.co/.test(currentNetwork.url);
```

Comment:
why are you using a different case convention here? 🤔   & could this regex be moved into a helper function with a descriptive name of what it does?

Reply:
This was to use the shorthand 
  const defaultProperties = {
    network: currentNetwork.name.toLowerCase(),
    using_default_hiro_api,
  };
I just fixed it.

Reply:
Actually I will use camelCase also in the json to stay consistent with segment convention.